diff --git a/src/model/model_utils.py b/src/model/model_utils.py
index 44b403a..8a37a43 100644
--- a/src/model/model_utils.py
+++ b/src/model/model_utils.py
@@ -11,9 +11,9 @@ model_dirs = {
     "falcon3-1b": "tiiuae/falcon3-1b-instruct",
     "falcon3-7b": "tiiuae/falcon3-7b-instruct",
     'qwen2.5-0.5b': 'Qwen/Qwen2.5-0.5B-Instruct',
-    'qwen2.5-1.5b': 'Qwen/Qwen2.5-1.5B-Instruct',
+    'qwen2.5-1.5b': '/mnt/quarkfs/share_model/Qwen2.5-1.5B',
     'qwen2.5-3b': 'Qwen/Qwen2.5-3B-Instruct',
-    'qwen2.5-7b': 'Qwen/Qwen2.5-7B-Instruct',
-    'qwen2.5-14b': 'Qwen/Qwen2.5-14B-Instruct',
+    'qwen2.5-7b': '/mnt/quarkfs/share_model/Qwen2.5-7B-Instruct',
+    'qwen2.5-14b': '/mnt/quarkfs/share_model/Qwen2.5-14B-Instruct',
     'qwen2.5-32b': 'Qwen/Qwen2.5-32B-Instruct'
 }
